torchrl/envs/transforms/transforms.py

```diff
@@ -4634,22 +4692,39 @@ def _reset(
         )
         _reset = _get_reset(self.reset_key, tensordict)
         if _reset.any():
-            for key, spec in self.primers.items():
+            for key, spec in self.primers.items(True, True):
                 if self.random:
                     value = spec.rand(shape)
                 else:
-                    value = torch.full_like(
-                        spec.zero(shape),
-                        self.default_value,
-                    )
-                prev_val = tensordict.get(key, 0.0)
-                value = torch.where(expand_as_right(_reset, value), value, prev_val)
+                    if isinstance(self.default_value, dict):
+                        value = self.default_value[key]
+                    else:
+                        value = self.default_value
+                    if callable(value):
+                        value = value()
+                        if not self._validated:
+                            self._validate_value_tensor(value, spec)
+                    else:
+                        value = torch.full_like(
+                            spec.zero(shape),
+                            value,
+                        )
+                        prev_val = tensordict.get(key, 0.0)
+                        value = torch.where(
+                            expand_as_right(_reset, value), value, prev_val
+                        )
                 tensordict_reset.set(key, value)
+            self._validated = True
         return tensordict_reset
 
     def __repr__(self) -> str:
         class_name = self.__class__.__name__
-        return f"{class_name}(primers={self.primers}, default_value={self.default_value}, random={self.random})"
+        default_value = (
+            self.default_value
+            if isinstance(self.default_value, float)
+            else self.default_value.__class__.__name__
+        )
```

Comment:
Maybe always a dict like {key: val}
And if val is a callable we just write `Callable` instead